lib/template.js

```diff
@@ -146,6 +146,10 @@ JsonTemplate.prototype.build = function(parameters) {
             required = true;
           }
           var val = parameters[name]; // Name is param[1]
+          // If there is a literal object in the string that has no schema,
+          // that probably means that it is intended as an object in the string.
+          // So here we'll update the index, and skip that object.
+          index = match.index;
```

Comment:
I don't really like the ambiguity. We should either introduce an `escaping` mechanism such as  `{{` for `{` or report that a variable cannot be resolved.

Reply:
Hi @raymondfeng , thank you for reviewing it. I could add a test case, but it would be exactly the same as `'should support object variables with expressions {var} when is not defined in template'`, so instead, I have just fixed the existing test.

I realized that the explanation in the comment is quite ambiguous, I'll update the comment again. For the new mechanism, I think we need one that can accept the expression (eg: `{var}`) as it is within the input when it is not defined, so maybe not reporting that the variable cannot be resolved as it is intended?